gobblin-temporal/src/main/java/org/apache/gobblin/temporal/joblauncher/GobblinTemporalJobLauncher.java

```diff
@@ -188,4 +191,17 @@ protected void removeTasksFromCurrentJob(List<String> workUnitIdsToRemove) {
   protected void addTasksToCurrentJob(List<WorkUnit> workUnitsToAdd) {
     log.warn("NOT IMPLEMENTED: Temporal addTasksToCurrentJob");
   }
+
+  @Override
+  public void close() throws IOException {
+    try {
+      // Calling cancel before calling close on serviceStubs as it will shutdown the service which is required during cancellation.
+      cancelJob(jobListener);
+    } catch (Exception e) {
+      log.error("Exception occurred while cancelling job", e);
+    } finally {
+      managedWorkflowServiceStubs.close();
```

Comment:
can this throw an exception? If so, close doesn't get called. 
looping twice with try/finally might be the correct way to do this.